packages/react-components/react-skeleton/src/components/Skeleton/useSkeletonStyles.ts

```diff
@@ -1,33 +1,134 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { SkeletonSlots, SkeletonState } from './Skeleton.types';
 import type { SlotClassNames } from '@fluentui/react-utilities';
+import { tokens } from '@fluentui/react-theme';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
 
 export const skeletonClassNames: SlotClassNames<SkeletonSlots> = {
   root: 'fui-Skeleton',
-  // TODO: add class names for all slots on SkeletonSlots.
-  // Should be of the form `<slotName>: 'fui-Skeleton__<slotName>`
+};
+
+const skeletonWaveAnimationRTL = {
+  from: {
+    backgroundPositionX: '300% /* @noflip */',
+  },
+  to: {
+    backgroundPositionX: '0% /* @noflip */',
+  },
+};
+
+const skeletonWaveAnimation = {
+  from: {
+    backgroundPositionX: '0% /* @noflip */',
+  },
+  to: {
+    backgroundPositionX: '300% /* @noflip */',
+  },
+};
+
+const skeletonPulseAnimation = {
+  from: {
+    opacity: '1',
+  },
+  to: {
+    opacity: '0.4',
+  },
 };
 
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
+const useRootStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    '& .fui-SkeletonItem': {
+      position: 'relative',
+      ...shorthands.overflow('hidden'),
+      backgroundSize: '300% 100%',
+      backgroundPositionX: 'center',
+      backgroundPositionY: 'center',
+      backgroundAttachment: 'fixed',
+      animationIterationCount: 'infinite',
+      animationDuration: '3s',
+      animationTimingFunction: 'linear',
+    },
```

Comment:
After the latest refactor, I think that it's even more reasonable to use the approach that @ling1726 suggested.

- `Skeleton` still accepts the same props
- `Skeleton` passes props that affect styles down via a context (React contexts are cheap 😊)
- `SkeletonItem` defines these styles, just move them to the component
- As a result, we have smaller bundle size and CSS rules that can be reused by other components 🔥

Reply:
I've updated the implementation to use context!